charmcraft/commands/store/client.py

```diff
@@ -43,12 +45,42 @@
 API_BASE_URL = 'https://api.staging.charmhub.io'
 STORAGE_BASE_URL = 'https://storage.staging.snapcraftcontent.com'
 
+TESTING_ENV_PREFIXES = ["TRAVIS", "AUTOPKGTEST_TMP"]
+
+
+def _get_os_platform(filepath=pathlib.Path("/etc/os-release")):
+    """Determine a system/release combo for an OS using /etc/os-release if available."""
+    system = platform.system()
+    release = platform.release()
+    machine = platform.machine()
+
+    if system == "Linux":
+        os_release = {}
+        try:
+            with filepath.open("r", encoding='utf-8') as f:
+                for line in f:
+                    if "=" in line:
+                        key, value = line.rstrip().split("=", 1)
+                        os_release[key] = value.strip('"')
+        except FileNotFoundError:
+            logger.debug("Unable to locate 'os-release' file, using default values")
+        finally:
+            system = os_release.get("NAME", "Unknown")
+            release = os_release.get("VERSION_ID", "Unknown Version")
```

Comment:
I think it's better, in case of not having `NAME` or `VERSION_ID`, to just leave original `system` and `release` (from `platform`, thank "Unknown"s.

IOW, what do you think about doing?

            system = os_release.get("NAME", system)
            release = os_release.get("VERSION_ID", release)

Reply:
No worries, happy to make changes. I emulated what Snapcraft was doing (https://github.com/snapcore/snapcraft/blob/b5138e03e53727347f65b0b1499a02fccd1ef9b4/snapcraft/storeapi/_agent.py#L35) but I agree that this is more clear. I have pushed an update!